workflow.go

```diff
@@ -82,14 +89,17 @@ func (w *Workflow) addStep(phase Phase, step Steper, config *StepConfig) {
 	if step == nil {
 		return
 	}
+	if stepToBuild, ok := step.(interface{ Build() }); ok {
```

Comment:
It is tricky. I think it should be not only mentioned in example, but also in the lib comment somewhere (in case for doc generation).

Reply:
this is kinda experimental right now... I'll trying to use it and collects feedback until the design is finished

Reply:
will remove it in next commit